src/agent/src/mount.rs

```diff
@@ -778,11 +778,18 @@ pub async fn add_storages(
             }
         };
 
-        // Todo need to rollback the mounted storage if err met.
-        let mount_point = res?;
-
-        if !mount_point.is_empty() {
-            mount_list.push(mount_point);
+        match res {
+            Ok(mount_point) => {
+                if !mount_point.is_empty() {
+                    mount_list.push(mount_point);
+                }
+            }
+            Err(e) => {
+                for m in mount_list {
```

Comment:
umount() may fail if mount points overlap, resulting in a silent mount point leak. Rollback should always undo things in reverse order to avoid this.